packages/front-end/components/ProjectBadges.tsx

```diff
@@ -1,18 +1,24 @@
 import clsx from "clsx";
+import { useState } from "react";
+import { FaInfoCircle } from "react-icons/fa";
 import { useDefinitions } from "@/services/DefinitionsContext";
 import Badge from "@/components/Badge";
+import Tooltip from "./Tooltip/Tooltip";
 
 export interface Props {
+  resourceType: "metric" | "data source" | "member" | "team" | "fact table";
   projectIds?: string[];
   sort?: boolean;
   className?: string;
 }
 
 export default function ProjectBadges({
+  resourceType,
   projectIds,
   sort = true,
   className = "badge-ellipsis short",
 }: Props) {
+  const [showMissingProjectError, setShowMissingProjectError] = useState(false);
```

Comment:
We don't really need a state variable for this, it could be calcculated on the fly:
```
const showMissingProjectErr = filteredProjects.some(p => !p);
```